src/middlewares/task.middleware.js

```diff
@@ -191,27 +192,36 @@ export default (store: any) => (next: any) => (action: any) => {
           )
         );
       } else {
-        // Info: No `shell: true` used so it's OK to kill just the childProcess.
-        childProcess.spawn('kill', ['-9', processId]);
-
-        ipcRenderer.send('removeProcessId', processId);
-
-        // Once the task is killed, we should dispatch a notification
-        // so that the terminal shows something about this update.
-        // My initial thought was that all tasks would have the same message,
-        // but given that we're treating `start` as its own special thing,
-        // I'm realizing that it should vary depending on the task type.
-        // TODO: Find a better place for this to live.
-        const abortMessage = isDevServerTask(name)
-          ? 'Server stopped'
-          : 'Task aborted';
-
-        next(
-          receiveDataFromTaskExecution(
-            task,
-            `\u001b[31;1m${abortMessage}\u001b[0m`
-          )
-        );
+        // Child node processes will persist after their parent's death
+        // if they are not killed first, so we need to use `psTree` to build
+        // a tree of children and kill them that way.
+        psTree(processId, (err, children) => {
```

Comment:
@superhawk610 Thanks for re-adding this and for adding the comment. I wasn't aware of this.
Can you please check `main.js` process killing and add child killing there as well?

The task killing during app run is working (tested with Win & Ubunutu). But the task killing doesn't work if the app is closed with a running dev server. PID still active - just tested with Ubuntu.